### ISSUE TITLE
RequiredShortTernaryOperator false positive

### ISSUE THREAD

**Issue by farhadsakhaei** [slevomat/coding-standard#1695]
Hi,
I think I get a false positive error:

**My Code:**
$login_link_text  = get_theme_mod( 'login_link_text' ) ? get_theme_mod( 'login_link_text' ) : 'Login';

**Error:**
[{
	"resource": "popup-login.php",
	"owner": "php",
	"code": "SlevomatCodingStandard.ControlStructures.RequireShortTernaryOperator.RequiredShortTernaryOperator",
	"severity": 8,
	"message": "Use short ternary operator.",
	"source": "PHPCS",
}]

This is not equivalent to :
$login_link_text  = get_theme_mod( 'login_link_text' ) ?? 'Login';

There is a slight difference in their syntax.

### 1. The first statement uses the ternary operator:
```php
$login_link_text = get_theme_mod('login_link_text') ? get_theme_mod('login_link_text') : 'Login';
```
This checks if `get_theme_mod('login_link_text')` is truthy. If it is, it assigns the value of `get_theme_mod('login_link_text')` to `$login_link_text`; otherwise, it assigns `'Login'`.

### 2. The second statement uses the null coalescing operator (`??`):
```php
$login_link_text = get_theme_mod('login_link_text') ?? 'Login';
```
This checks if `get_theme_mod('login_link_text')` is `null`. If it is `null`, it assigns `'Login'`; otherwise, it assigns the value of `get_theme_mod('login_link_text')`.

### Key Difference:
- The **ternary operator** checks if the value is truthy (i.e., evaluates to true in a boolean context, which means it could fail if the value is `0`, `false`, `''`, etc.).
- The **null coalescing operator** only checks if the value is strictly `null`.

If you want to handle only `null` values, the null coalescing operator (`??`) is preferable.
If you need to check for any falsy value, including `false`, `0`, or `''`, then the ternary operator is better suited.

**Comment by farhadsakhaei** [comment]
It's ok !!!
